homeassistant/components/media_player/kodi.py

```diff
@@ -218,94 +231,124 @@ def supported_media_commands(self):
 
         return supported_media_commands
 
-    def turn_off(self):
+    @asyncio.coroutine
+    def async_turn_off(self):
         """Execute turn_off_action to turn off media player."""
         if self._turn_off_action == 'quit':
-            self._server.Application.Quit()
+            yield from self._server.Application.Quit()
         elif self._turn_off_action == 'hibernate':
-            self._server.System.Hibernate()
+            yield from self._server.System.Hibernate()
         elif self._turn_off_action == 'suspend':
-            self._server.System.Suspend()
+            yield from self._server.System.Suspend()
         elif self._turn_off_action == 'reboot':
-            self._server.System.Reboot()
+            yield from self._server.System.Reboot()
         elif self._turn_off_action == 'shutdown':
-            self._server.System.Shutdown()
+            yield from self._server.System.Shutdown()
         else:
             _LOGGER.warning('turn_off requested but turn_off_action is none')
 
-        self.update_ha_state()
+        yield from self.async_update_ha_state()
```

Comment:
That makes more sense than what was there. I converted it since it was there originally. Happy to get rid of them.